### ISSUE TITLE
Trying to edit a table cell causes infinite spam of CellEditEvents

### ISSUE THREAD

**Issue by zilti** [cljfx/cljfx#59]
As the title says. I created a self-contained example to demonstrate it: https://gist.github.com/zilti/7602e7c099c491734e2df9cf11e72fb4

Launch it, then double-click a table cell to see it happen.

**Comment by vlaaad** [comment]
Hi!
Thanks for the repro, I was able to reproduce the issue. I haven't figured out yet what's causing it, but I remember similar painful experiences when trying to use editable cells from cljfx. In the end I just reimplemented editable views myself in code; there is [an example](https://github.com/cljfx/cljfx/blob/master/examples/e32_editable_table_cell_with_datascript.clj) doing that.

**Comment by vlaaad** [comment]
Note: I made event handling synchronous to see the full stack trace, but it does not seem to be hinting at something in particular:
```
tap> #reveal/error{:via [{:type java.lang.RuntimeException
                          :at [user$eval33958$fn__33960 invoke "user.clj" 204]}]
                   :trace [[user$eval33958$fn__33960 invoke "user.clj" 204]
                           [clojure.lang.MultiFn invoke "MultiFn.java" 229]
                           [cljfx.event_handler$wrap_co_effects$fn__33690 invoke "event_handler.clj" 10]
                           [cljfx.event_handler$wrap_effects$dispatch_sync_BANG___33703 invoke "event_handler.clj" 27]
                           [cljfx.event_handler$process_event invokeStatic "event_handler.clj" 31]
                           [cljfx.event_handler$process_event invoke "event_handler.clj" 30]
                           [clojure.lang.AFn applyToHelper "AFn.java" 171]
                           [clojure.lang.AFn applyTo "AFn.java" 144]
                           [clojure.core$apply invokeStatic "core.clj" 671]
                           [clojure.core$binding_conveyor_fn$fn__5754 doInvoke "core.clj" 2041]
                           [clojure.lang.RestFn applyTo "RestFn.java" 146]
                           [clojure.lang.Agent$Action doRun "Agent.java" 114]
                           [clojure.lang.Agent$Action run "Agent.java" 163]
                           [user$reify__42318 execute "user.clj" 209]
                           [clojure.lang.Agent$Action execute "Agent.java" 90]
                           [clojure.lang.Agent enqueue "Agent.java" 268]
                           [clojure.lang.Agent dispatchAction "Agent.java" 255]
                           [clojure.lang.Agent dispatch "Agent.java" 241]
                           [clojure.core$send_via invokeStatic "core.clj" 2109]
                           [clojure.core$send_via doInvoke "core.clj" 2101]
                           [clojure.lang.RestFn invoke "RestFn.java" 573]
                           [cljfx.event_handler$wrap_async$dispatch_async_BANG___33719 invoke "event_handler.clj" 43]
                           [cljfx.lifecycle$make_handler_fn$fn__33354 invoke "lifecycle.clj" 128]
                           [cljfx.coerce$event_handler$reify__32938 handle "coerce.clj" 135]
                           [com.sun.javafx.event.CompositeEventHandler dispatchBubblingEvent "CompositeEventHandler.java" 86]
                           [com.sun.javafx.event.EventHandlerManager dispatchBubblingEvent "EventHandlerManager.java" 238]
                           [com.sun.javafx.event.EventHandlerManager dispatchBubblingEvent "EventHandlerManager.java" 191]
                           [com.sun.javafx.event.BasicEventDispatcher dispatchEvent "BasicEventDispatcher.java" 58]
                           [com.sun.javafx.event.EventDispatchChainImpl dispatchEvent "EventDispatchChainImpl.java" 114]
                           [com.sun.javafx.event.EventUtil fireEventImpl "EventUtil.java" 74]
                           [com.sun.javafx.event.EventUtil fireEvent "EventUtil.java" 54]
                           [javafx.event.Event fireEvent "Event.java" 198]
                           [javafx.scene.control.TableCell startEdit "TableCell.java" 332]
                           [cljfx.fx.table_cell.proxy$javafx.scene.control.TableCell$ff19274a startEdit nil -1]
                           [javafx.scene.control.TableCell updateEditing "TableCell.java" 558]
                           [javafx.scene.control.TableCell indexChanged "TableCell.java" 474]
                           [javafx.scene.control.IndexedCell updateIndex "IndexedCell.java" 120]
                           [cljfx.fx.table_cell.proxy$javafx.scene.control.TableCell$ff19274a updateIndex nil -1]
                           [javafx.scene.control.skin.TableRowSkinBase updateCells "TableRowSkinBase.java" 539]
                           [javafx.scene.control.skin.TableRowSkinBase <init> "TableRowSkinBase.java" 159]
                           [javafx.scene.control.skin.TableRowSkin <init> "TableRowSkin.java" 89]
                           [javafx.scene.control.TableRow createDefaultSkin "TableRow.java" 213]
                           [javafx.scene.control.Control doProcessCSS "Control.java" 897]
                           [javafx.scene.control.Control access$000 "Control.java" 83]
                           [javafx.scene.control.Control$1 doProcessCSS "Control.java" 89]
                           [com.sun.javafx.scene.control.ControlHelper processCSSImpl "ControlHelper.java" 67]
                           [com.sun.javafx.scene.NodeHelper processCSS "NodeHelper.java" 147]
                           [javafx.scene.Node processCSS "Node.java" 9515]
                           [javafx.scene.Node applyCss "Node.java" 9602]
                           [javafx.scene.control.skin.VirtualFlow setCellIndex "VirtualFlow.java" 1715]
                           [javafx.scene.control.skin.VirtualFlow addTrailingCells "VirtualFlow.java" 2042]
                           [javafx.scene.control.skin.VirtualFlow layoutChildren "VirtualFlow.java" 1253]
                           [javafx.scene.Parent layout "Parent.java" 1211]
                           [javafx.scene.Parent layout "Parent.java" 1218]
                           [javafx.scene.Parent layout "Parent.java" 1218]
                           [javafx.scene.Scene doLayoutPass "Scene.java" 590]
                           [javafx.scene.Scene$ScenePulseListener pulse "Scene.java" 2507]
                           [com.sun.javafx.tk.Toolkit lambda$runPulse$2 "Toolkit.java" 410]
                           [java.security.AccessController doPrivileged "AccessController.java" -2]
                           [com.sun.javafx.tk.Toolkit runPulse "Toolkit.java" 409]
                           [com.sun.javafx.tk.Toolkit firePulse "Toolkit.java" 436]
                           [com.sun.javafx.tk.quantum.QuantumToolkit pulse "QuantumToolkit.java" 524]
                           [com.sun.javafx.tk.quantum.QuantumToolkit pulse "QuantumToolkit.java" 504]
                           [com.sun.javafx.tk.quantum.QuantumToolkit pulseFromQueue "QuantumToolkit.java" 497]
                           [com.sun.javafx.tk.quantum.QuantumToolkit lambda$runToolkit$11 "QuantumToolkit.java" 320]
                           [com.sun.glass.ui.InvokeLaterDispatcher$Future run "InvokeLaterDispatcher.java" 96]]}
```

**Comment by vlaaad** [comment]
I investigated this a bit more. 

My current lead is `detached-prop-map`: when the description or opts change (which is the case of every context update), it re-creates cells from scratch instead of updating them. I think re-creating cells triggers the re-firing of the start-edit event. Proper solution to this would be somehow accessing all created cells and updating them when the description changes instead of re-creating a cell factory that actualizes the view.

Your immediate problem is worked around by not updating context when it's already edited:
```clj
  (when (and (instance? javafx.scene.control.TableColumn$CellEditEvent event)
             (not (fx/sub context :edit)))
    {:context (fx/swap-context context assoc :edit [attr (::index (.getRowValue event))])})
```

I'm not sure the current situation won't create problems on your next step with editable cells, I would still suggest not using JavaFX-provided cell editing and instead reimplement it as in the example above.

**Comment by vlaaad** [comment]
This problem is fixed in [1.7.1](https://github.com/cljfx/cljfx/blob/master/CHANGELOG.md#171---2020-05-10): you should define `:cell-factory` like that:
```clj
{:fx/cell-type :table-cell
 :describe (editable-cell-factory editable-cell data attr :default)}
```
